lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2386,12 +2386,18 @@ public function getCustomTypeDeclarationSQL(array $columnDef)
      * Obtains DBMS specific SQL code portion needed to set an index
      * declaration to be used in statements like CREATE TABLE.
      *
-     * @param array $fields
+     * @param array|Index $fields
```

Comment:
I know the API currently accepts an array of field names. My first intention was to change this to only accept an `Index` parameter. I agree that an `IndexColumn` could be a solution but it seems heavier to implement this class.

I did not wanted to have a simple array of lengths in the API but I currently do not see another better way to avoid this while keeping BC.